tests/jenkins/BuildPipelineTest.groovy

```diff
@@ -51,6 +52,8 @@ abstract class BuildPipelineTest extends CommonPipelineTest {
         binding.setVariable('scm', {})
 
         helper.registerAllowedMethod("legacySCM", [Closure.class], null)
```

Comment:
Shall we remove legacy SCM since we dont use it anymore?

Reply:
yep .. I think so ... as we are not going back to use legacySCM anymore ... let me go remove it and rerun the test .. thanks

Reply:
Looks like we cannot remove this helper.registerAllowedMethod("legacySCM", [Closure.class], null) at he moment yet although it does load shared lib from opensearch-build-libraries.git 

I think when the job downloads the remote lib (via the modernSCM method) from the opensearch-build-libraries repo into the local, and then finds the lib version is with "20211123" (for now), it then requires the legacySCM method to process.

Maybe we can keep the legacySCM regsiterAllowMethod for now, and come back to clean it up after we solve https://github.com/opensearch-project/opensearch-build-libraries/issues/7 ?  Thanks.

@peterzhuamazon @gaiksaya @prudhvigodithi